src/tsd-metrics.ts

```diff
@@ -0,0 +1,359 @@
+/*
+ * Copyright 2016 Groupon.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import tsdDef = require("tsdDef");
+import events = require("events");
+import _ = require("underscore");
+
+import timers = require("./tsd-timer");
+import counters = require("./tsd-counter");
+import utils = require("./utils");
+import timerSamples = require("./timer-samples");
+import counterSamples = require("./counter-samples");
+import units = require("./tsd-units");
+import sample = require("./tsd-metric-sample");
+import metricsList = require("./tsd-metrics-list");
+import metricsEvent = require("./tsd-metrics-event");
+import metricsFactory = require("./tsd-metrics-factory");
+import log4jsSink = require("./sinks/tsd-query-log-sink");
+import consoleSink = require("./sinks/tsd-console-sink");
+import sink = require("./sinks/tsd-sink");
+import errors = require("./error-reporting");
+
+//aliases
+import TsdTimer = timers.TsdTimer;
+import TsdCounter = counters.TsdCounter;
+import Lazy = utils.Lazy;
+import CounterSamples = counterSamples.CounterSamples;
+import TimerSamples = timerSamples.TimerSamples;
+import Units = units.Units;
+import TsdMetricSample = sample.TsdMetricSample;
+import TsdMetricsList = metricsList.TsdMetricsList;
+import TsdMetricsEvent = metricsEvent.TsdMetricsEvent;
+import TsdMetricsFactory = metricsFactory.TsdMetricsFactory
+
+/**
+ * Node JS class for publishing metrics as time series data (TSD).
+ *
+ * @class
+ * @author Mohammed Kamel (mkamel at groupon dot com)
+ */
+export class TsdMetrics implements tsdDef.Metrics {
+
+    /**
+     * Metrics state object that holds flag to the metrics open state along with common event emitter.
+     */
+    private _metricsStateObject:MetricsStateObject = new MetricsStateObject();
+    private _timers:{[name:string]: tsdDef.Timer} = {};
+    private _counters:{[name:string]: tsdDef.Counter} = {};
```

Comment:
*PUNT* This code is just moved from `src/tsd-metrics-client.ts`, but I can look into it if you really don't believe that the tests are doing their job.